packages/@aws-cdk/aws-secretsmanager/lib/secret.ts

```diff
@@ -0,0 +1,259 @@
+import iam = require('@aws-cdk/aws-iam');
+import kms = require('@aws-cdk/aws-kms');
+import cdk = require('@aws-cdk/cdk');
+import secretsmanager = require('./secretsmanager.generated');
+
+/**
+ * A secret in AWS Secrets Manager.
+ */
+export interface ISecret extends cdk.IConstruct {
+  /**
+   * The customer-managed encryption key that is used to encrypt this secret, if any. When ``unknown``, the default KMS
+   * key for the account and region is being used.
+   */
+  readonly encryptionKey?: kms.IEncryptionKey;
+
+  /**
+   * The ARN of the secret in AWS Secrets Manager.
+   */
+  readonly secretArn: string;
+
+  /**
+   * Exports this secret.
+   *
+   * @return import props that can be passed back to ``Secret.import``.
+   */
+  export(): SecretImportProps;
+
+  /**
+   * Grants reading the secret value to some role.
+   *
+   * @param grantee       the principal being granted permission.
+   * @param versionStages the version stages the grant is limited to. If not specified, no restriction on the version
+   *                      stages is applied.
+   */
+  grantRead(grantee: iam.IPrincipal, versionStages?: string[]): void;
+}
+
+/**
+ * The properties required to create a new secret in AWS Secrets Manager.
+ */
+export interface SecretProps {
+  /**
+   * An optional, human-friendly description of the secret.
+   */
+  description?: string;
+
+  /**
+   * The customer-managed encryption key to use for encrypting the secret value.
+   *
+   * @default a default KMS key for the account and region is used.
+   */
+  encryptionKey?: kms.IEncryptionKey;
+
+  /**
+   * The value of the secret. Either this or ``generateSecretString`` must be specified, but not both.
+   *
+   * @default a secret is generated using ``generateSecretString`` configuration.
+   */
+  secretString?: string;
```

Comment:
Does this mean that users would have clear secret strings in their CDK apps/source code?
Sounds like something we want to protect people from doing...

Reply:
They could also wire in a parameter, for example. I agree this is the sign of poor practices, but there are things you *have* to do like that (create with a placeholder, then manually replace the value)... This is L2, so opinions are irrelevant :)

Reply:
I'm adding some verbiage in the documentation block though.

Reply:
I think we should stray from our "non opinionated" tenet to protect the security of our customers (the First Law of Robotics). How about this? Let's omit this feature from our API for the first version and see if we get customer feedback that it's needed?